vision/interface.py

```diff
@@ -0,0 +1,51 @@
+#rectangle object
```

Comment:
You also need an official unit tests in vision/unit_tests that follows the description given in vision/README.md.

You should give the Environment object a bunch of data normal and some that you think could break it - ensure it works as it should.

Reply:
Make sure you include a test where the length of env.rects changes during iteration.

-- also storing self.length is unnecessary because this info is already stored by the interpreter and is just referenced when calling len(). Replace i %= self.length with i %= len(self.rects) and delete other occurrences of self.length.
https://stackoverflow.com/questions/2399835/in-python-what-does-lenlist-do#2399932